app/services/aws_roles.py

```diff
@@ -44,21 +45,68 @@ def oidc_arn(self, domain: str) -> str:
         resource = f"oidc-provider/{domain}"
         return self.build_arn(service=self.SERVICE, resource=resource)
 
-    def get_trust_policy_data(self) -> dict[str, str]:
+    def get_trust_policy_data(self, oidc_user_id: str) -> dict[str, str]:
         template = self.jinja.get_template("roles/trust_policy.json")
         context = {
-            "username": self.username,
+            "username": self.rolename,
             "oidc_arn": self.oidc_arn(settings.oidc_domain),
             "oidc_domain": settings.oidc_domain,
-            "oidc_user_id": self.oidc_user_id,
+            "oidc_user_id": oidc_user_id,
             "eks_arn": self.oidc_arn(settings.oidc_eks_provider),
             "oidc_eks_provider": settings.oidc_eks_provider,
         }
         rendered = template.render(**context)
         return json.loads(rendered)
 
-    def create_role(self):
-        trust_policy = self.get_trust_policy_data()
+    def create_role(self, oidc_user_id):
+        trust_policy = self.get_trust_policy_data(oidc_user_id=oidc_user_id)
         return self.client.create_role(
-            RoleName=self.username, AssumeRolePolicyDocument=json.dumps(trust_policy)
+            RoleName=self.rolename, AssumeRolePolicyDocument=json.dumps(trust_policy)
         )
+
+    def get_policies_for_role(self) -> list[dict]:
```

Comment:
Could consider adding a unit tests for this method - although not a big deal as there is a test for the `get_policies_by_role_name` which ultimately checks the policies returned

Reply:
Now included